horovod/run/common/util/safe_shell_exec.py

```diff
@@ -69,75 +74,104 @@ def prepend_context(line, rank, prefix):
             line=line
         )
 
-    with os.fdopen(src_fd, 'r') as src:
-        line_buffer = ''
-        while True:
-            text = os.read(src.fileno(), 1000)
-            if not isinstance(text, str):
-                text = text.decode('utf-8')
-            if not text:
-                break
+    line_buffer = ''
+    while True:
+        text = os.read(src_stream.fileno(), 1000)
+        if not isinstance(text, str):
+            text = text.decode('utf-8')
+        if not text or len(text) == 0:
```

Comment:
`not text` tests for `None` or empty, right? `len(text) == 0` tests for empty again, maybe better to be explicit what `not text` is expected to test

Reply:
Good point.  Changed to check for None before the decode, and remove the length check here.